### ISSUE TITLE
MATLink on Mathematica v10

### ISSUE THREAD

**Issue by tjof2** [rsmenon/MATLink#73]
Just installed Mathematica v10 on Ubuntu 12.04, with MATLAB r2013b. 

Running the first example from the website, I get the following:

![temp](https://cloud.githubusercontent.com/assets/6626247/3549737/409a601c-08ce-11e4-9eb3-24ea0b99b337.png)

It was working fine before on v9, I've not changed anything.

I then proceeded to recompile mengine manually, and got the following compiler error:

```
/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions/mprep -prototypes mengine.tm > menginetm.cpp
g++ -O -Wall -I/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions -I/usr/local/MATLAB/R2013b/bin/../extern/include -L/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions -L/usr/local/MATLAB/R2013b/bin/../bin/glnxa64 menginetm.cpp mengine.cpp set.cpp get.cpp main.cpp -Wl,-rpath-link,/usr/local/MATLAB/R2013b/bin/../bin/glnxa64/ -lML64i3 -lmx -leng -o mengine -lrt -pthread
set.cpp: In function ‘void eng_make_String(const short unsigned int*, int, int)’:
set.cpp:159:31: warning: narrowing conversion of ‘len’ from ‘int’ to ‘mwSize {aka long unsigned int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
set.cpp: In function ‘void eng_make_Struct()’:
set.cpp:191:5: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions/libML64i3.so: error: undefined reference to 'uuid_parse'
/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions/libML64i3.so: error: undefined reference to 'uuid_generate'
/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions/libML64i3.so: error: undefined reference to 'uuid_unparse_upper'
/tmp/cciuFtXx.o:menginetm.cpp:function MLAnswer: error: undefined reference to 'MLWaitForLinkActivity'
collect2: error: ld returned 1 exit status
make: *** [mengine] Error 1
:~/.Mathematica/Applications/MATLink/Engine/src$ sudo make -f Makefile.lin64
g++ -O -Wall -I/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions -I/usr/local/MATLAB/R2013b/bin/../extern/include -L/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions -L/usr/local/MATLAB/R2013b/bin/../bin/glnxa64 menginetm.cpp mengine.cpp set.cpp get.cpp main.cpp -Wl,-rpath-link,/usr/local/MATLAB/R2013b/bin/../bin/glnxa64/ -lML64i3 -lmx -leng -o mengine -lrt -pthread
set.cpp: In function ‘void eng_make_String(const short unsigned int*, int, int)’:
set.cpp:159:31: warning: narrowing conversion of ‘len’ from ‘int’ to ‘mwSize {aka long unsigned int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
set.cpp: In function ‘void eng_make_Struct()’:
set.cpp:191:5: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions/libML64i3.so: error: undefined reference to 'uuid_parse'
/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions/libML64i3.so: error: undefined reference to 'uuid_generate'
/usr/local/Wolfram/Mathematica/10.0/Executables/../SystemFiles/Links/MathLink/DeveloperKit/Linux-x86-64/CompilerAdditions/libML64i3.so: error: undefined reference to 'uuid_unparse_upper'
/tmp/ccyTui44.o:menginetm.cpp:function MLAnswer: error: undefined reference to 'MLWaitForLinkActivity'
collect2: error: ld returned 1 exit status
make: *** [mengine] Error 1
```

**Comment by szhorvat** [comment]
It will take a while until I'll be able to test this on Linux.

Until then, please try this:

Find Makefile.lin64 in the MATLink directory tree, open it, and change `ML64i3` to `ML64i4`.  Then try again.  (Do run `make -f Makefile.lin64 clean` first...) Let me know if it helped.

Once compiled, MATLink should work fine with Mathematica 10.  We ran the test suite with Mathematcia 10 on OS X, which is a very similar platform to Linux from MATLink's perspective.

Note: Now you have MATLAB R2013b.  Please avoid upgrading to R2014a if you use MATLink.  There's a good chance that MATLink won't work with R2014a on Linux (on OS X it doesn't). We're waiting for a MATLAB bug to be fixed in R2014b, as working around this bug would be very difficult.

**Comment by szhorvat** [comment]
Solution: since Mathematica 10, it is necessary to link with `-luuid` on Linux.

**Comment by szhorvat** [comment]
Fixed in the makefile for MATLink 1.1.
